feat(rfp): alternative javascript console for polkadot js API (#1545)
w3f/Grants-Program: docs/RFPs/Open/alternative-polkadot-js-api-console.md

# Alternative javascript console for Polkadot JS API

* **Status:** Open (anyone is allowed to apply)
* **Proposer:** [muddlebee](https://github.com/muddlebee)
* **Projects you think this work could be useful for** [optional]: Javascript console at https://polkadot.js.org/apps/#/js

## Project Description :page_facing_up: 

### Current state
We have a Javascript console at the Developer tab which is really useful for running polkadot js API scripts [polkadot JS API docs](https://polkadot.js.org/docs/)

**Link** - [https://polkadot.js.org/apps/#/js](https://polkadot.js.org/apps/#/js), UI screenshot below

![image](https://user-images.githubusercontent.com/8139783/197954316-1449075f-b8be-4a30-a759-873c15f8a14f.png)

**Current limitations of the above console**
Cannot save code properly, not much keyboard shortcuts, cannot customize configurations.

**Alternative polkadot js API playground**
1. [https://github.com/subdirectory/subshell](https://github.com/subdirectory/subshell)

### Proposed-RFP

> A new polkadot js API playground with vs-code like configurations like save the code, workspace, keyboard shortcuts etc etc..

Reviewer: ```suggestion
A new Polkadot-JS API playground with VS Code-like configurations like save the code, workspace, keyboard shortcuts, etc.
```

[https://polkadot.js.org/apps/#/js](https://polkadot.js.org/apps/#/js)

some examples -> https://playground.substrate.dev/
here we have to manually build and run our js bundles
![image](https://user-images.githubusercontent.com/8139783/198254152-abdd0f2e-62d4-4f0f-aad1-bcfdd6d48a74.png)

> **Why alternative javascript console for for polkadot js API**

Reviewer: ```suggestion
**Why alternative javascript console for Polkadot-JS API**?
```

Current polkadot js API console which I mentioned in beginning of this post, has some limitations, which we can overcome by creating a better version for smoother dev experience.

## Deliverables :nut_and_bolt:

The following items could be the initial deliverables of the project. Of course, improvements and additions are more than welcome.

- Initial research:
  - study how the current javascript console is developed at https://polkadot.js.org/apps/#/js
  - understand the libaries currently integrated from [polkadot JS API docs](https://polkadot.js.org/docs/)

- Development:
  - design a new UI/UX with better experience than current javascript console with features like
    - save code preferably with secure session management
    - keyboard shortcuts
    - [example](#Proposed-RFP)
    

>     more features which makes the polkadot js experience productive and smoother are welcome

Reviewer: ```suggestion
    Any additional features which make the Polkadot-JS experience more productive and smoother are welcome.
```
